libraries/chain/database.cpp

```diff
@@ -1451,6 +1451,45 @@ namespace golos { namespace chain {
             FC_CAPTURE_AND_RETHROW((to_account.name)(steem))
         }
 
+        /**
+         * @param to_account - the account to receive the new delegated vesting shares
+         * @param STEEM - STEEM to be converted to vesting shares
+         */
+        asset database::create_delegated_vesting(const account_object& to_account, asset steem) {
+            try {
+                const auto& cprops = get_dynamic_global_properties();
+
+                /**
+                 *  The ratio of total_vesting_shares / total_vesting_fund_steem should not
+                 *  change as the result of the user adding funds
+                 *
+                 *  V / C = (V+Vn) / (C+Cn)
+                 *
+                 *  Simplifies to Vn = (V * Cn ) / C
+                 *
+                 *  If Cn equals o.amount, then we must solve for Vn to know how many new vesting shares
+                 *  the user should receive.
+                 *
+                 *  128 bit math is requred due to multiplying of 64 bit numbers. This is done in asset and price.
+                 */
+                asset new_vesting = steem * cprops.get_vesting_share_price();
+
+                modify(to_account, [&](account_object& to) {
+                    to.delegated_vesting_shares += new_vesting;
```

Comment:
`delegated_vesting_shares` is part of `vesting_shares`, so you must first add to `vesting_shares`.
Maybe I missed some part of discussion, why did you decide to add separate `create_delegated_vesting` method? It fits good with standard `create_vesting`, something like:
```
asset claim_in_vesting = create_vesting(delegator, delegator_claim));
delegator.delegated_vesting_shares += claim_in_vesting;
```
Furthermore, `received_vesting_shares` of delegatee must be updated too.

Reply:
@zxcat OK, understood all, excluding the last paragraph.

Reply:
Each delegation change have 2 sides: delegator and delegatee.
1. delegated_vests field changes in delegator's account
2. received_vests field changes in delegatee's account